presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -0,0 +1,740 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
```

Comment:
Maybe it makes sense to move it to separate module? After all, this is a library that has no dependency on other Presto modules. It only uses Airlift.

The problem is that SPI exposes implementation of certain operations on types. Removing it from SPI will require pushing `TypeRegistry` or `FunctionRegistry` to places, where it wasn't used before. I tried it once, it's rather not trivial to refactor it like that.